src/index.mjs

```diff
@@ -65,22 +66,49 @@ export function listen(options) {
 
   const allowed = options.origins || [location.hostname];
   const ignores = options.ignores || [];
+  const delay = options.delay || 0;
+  const hrefsInViewport = [];
 
   const timeoutFn = options.timeoutFn || requestIdleCallback;
   const hrefFn = typeof options.hrefFn === 'function' && options.hrefFn;
 
+  const setTimeoutIfDelay = (callback, delay) => {
+    if (!delay) {
+      callback();
+      return;
+    }
+    setTimeout(callback, delay);
+  }
+
   const observer = new IntersectionObserver(entries => {
     entries.forEach(entry => {
+      // On enter
       if (entry.isIntersecting) {
-        observer.unobserve(entry = entry.target);
-        // Do not prefetch if will match/exceed limit
-        if (toPrefetch.size < limit) {
-          toAdd(() => {
-            prefetch(hrefFn ? hrefFn(entry) : entry.href, options.priority).then(isDone).catch(err => {
-              isDone(); if (options.onError) options.onError(err);
+        entry = entry.target;
+        // Adding href to array of hrefsInViewport
+        hrefsInViewport.push(entry.href);
+
+        // Setting timeout
+        setTimeoutIfDelay(() => {
+          const found = hrefsInViewport.indexOf(entry.href) > -1;
+          if (!found) return;
```

Comment:
If `found` is not being reused, it may be worth modifying this inline to if (! < the condition>) return; rather than defining a constant.

Reply:
Hey, thanks for the comment. Our intention was to make the code more readable, but it's your repo so we will change it.